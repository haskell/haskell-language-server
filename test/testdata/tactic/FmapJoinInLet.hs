{-# LANGUAGE ScopedTypeVariables #-}

fJoin :: forall f m a. (Monad m, Monad f) => f (m (m a)) -> f (m a)
fJoin =  let f = (_ :: m (m a) -> m a) in fmap f
