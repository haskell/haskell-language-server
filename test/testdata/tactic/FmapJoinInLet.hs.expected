{-# LANGUAGE ScopedTypeVariables #-}

fJoin :: forall f m a. (Monad m, Monad f) => f (m (m a)) -> f (m a)
fJoin =  let f = ( (\ mma -> (>>=) mma (\ ma -> ma)) :: m (m a) -> m a) in fmap f
